addons/website/static/src/js/set_view_track.js

```diff
@@ -0,0 +1,74 @@
+odoo.define('website.set_view_track', function (require) {
+"use strict";
+
+var CustomizeMenu = require('website.customizeMenu');
+var rpc = require('web.rpc');
+var Widget = require('web.Widget');
+
+var TrackPage = Widget.extend({
+    template: 'website.track_page',
+    xmlDependencies: ['/website/static/src/xml/track_page.xml'],
+    events: {
+        'change #switch-track-page': '_onTrackChange',
+    },
+    track: null,
+    start: function () {
+        var self = this;
+        this.$input = this.$('#switch-track-page');
+        this._is_tracked().then(function (data) {
+            if (data[0]['track']) {
+                self.track = true;
+                self.$input.attr('checked', 'checked');
+            } else {
+                self.track = false;
+            }
+        });
+    },
+    _is_tracked: function (val) {
+        var viewid = $('html').data('viewid');
+        if (!viewid) {
+            return Promise.reject();
+        } else {
+            return rpc.query({
+                model: 'ir.ui.view',
+                method: 'read',
+                args: [[viewid], ['track']],
+            });
+        }
+    },
+    _onTrackChange: function (ev) {
+        var checkboxValue = this.$input.is(':checked');
+        if (checkboxValue !== this.track) {
+            this.track = checkboxValue;
+            this._trackPage(checkboxValue);
+        }
+    },
+    _trackPage: function (val) {
+        var viewid = $('html').data('viewid');
+        if (!viewid) {
+            return Promise.reject();
+        } else {
+            return rpc.query({
+                model: 'ir.ui.view',
+                method: 'write',
+                args: [[viewid], {track: val}],
+            });
+        }
+    },
+});
+
+CustomizeMenu.include({
+    _loadCustomizeOptions: function () {
```

Comment:
the trackpage button is not a customize option, so the feature should not be added by extending this method. In fact, what was this ugly `__trackpageLoaded` variable even for? Why not doing this the same way as most navbar option (like ace editor, etc, etc): attach the widget to the track page button rendered server side. Most of the JS here would be gone I guess.

Reply:
I just moved this file from it's old enterprise module wich was deleted in another commit.  It did the job so I did not really look into it... laziness...

Reply:
Mmh, let's keep it and improve later if you do not have time to improve it now